v1/plugins/rest/azure.go

```diff
@@ -178,3 +184,104 @@ func buildAzureManagedIdentitiesRequestPath(
 
 	return endpoint + "?" + params.Encode()
 }
+
+type azureKeyVaultSignPlugin struct {
+	config  azureKeyVaultConfig
+	tokener func() (string, error)
+}
+
+func newKeyVaultSignPlugin(ap *azureManagedIdentitiesAuthPlugin, cfg *azureKeyVaultConfig) *azureKeyVaultSignPlugin {
+	resp := &azureKeyVaultSignPlugin{
+		tokener: func() (string, error) {
+			resp, err := azureManagedIdentitiesTokenRequest(
+				ap.Endpoint,
+				ap.APIVersion,
+				cfg.URL.String(),
+				ap.ObjectID,
+				ap.ClientID,
+				ap.MiResID,
+				ap.UseAppServiceMsi)
+			if err != nil {
+				return "", err
+			}
+			return resp.AccessToken, nil
+		},
+		config: *cfg,
+	}
+	return resp
+}
+
+func (akv *azureKeyVaultSignPlugin) setDefaults() {
+	if akv.config.APIVersion == "" {
+		akv.config.APIVersion = defaultKeyVaultAPIVersion
+	}
+}
+
+type kvRequest struct {
+	Alg   string `json:"alg"`
+	Value string `json:"value"`
+}
+
+type kvResponse struct {
+	KID   string `json:"kid"`
+	Value string `json:"value"`
+}
+
+// SignDigest() uses the Microsoft keyvault rest api to sign a byte digest
+// https://learn.microsoft.com/en-us/rest/api/keyvault/keys/sign/sign
+func (ap *azureKeyVaultSignPlugin) SignDigest(ctx context.Context, digest []byte) (string, error) {
+	tkn, err := ap.tokener()
+	if err != nil {
+		return "", err
+	}
+	if ap.config.URL.Host == "" {
+		return "", errors.New("keyvault host not set")
+	}
+
+	signingURL := ap.config.URL.JoinPath("keys", ap.config.Key, ap.config.KeyVersion, "sign")
+	q := signingURL.Query()
+	q.Set("api-version", ap.config.APIVersion)
+	signingURL.RawQuery = q.Encode()
+	reqBody, err := json.Marshal(kvRequest{
+		Alg:   ap.config.Alg,
+		Value: base64.StdEncoding.EncodeToString(digest)})
+	if err != nil {
+		return "", err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, signingURL.String(), bytes.NewBuffer(reqBody))
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+tkn)
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.Body != nil {
+			defer resp.Body.Close()
+			b, _ := io.ReadAll(resp.Body)
+			return "", fmt.Errorf("non 200 status code, got: %d. Body: %v", resp.StatusCode, string(b))
+		}
+		return "", fmt.Errorf("non 200 statuscode from keyvault sign, got: %d", resp.StatusCode)
```

Comment:
```suggestion
		return "", fmt.Errorf("non 200 status code from keyvault sign, got: %d", resp.StatusCode)
```

just to match the above error

Reply:
right, thank you for noticing 😄  Fixed in: [4540de4](https://github.com/open-policy-agent/opa/commit/4540de41e768efc6c3ee41c1b85e8a9cf64fb74c)